File: Problems/FixTheCode/FTC_3_sorting-algorithms/python/source/main.py
from bubbleSorter import BubbleSorter
from customSorter import CustomSorter 
from insertionSorter import InsertionSorter
from selection_sorter import SelectionSorter
from performance_test import PerformanceTest
import random

class SorterMain:

    def execute(self):
        self.execute_sorting_tests()
        self.execute_performance_tests()

    def print_array(self, array):
        array_str = ""
        for i in range(len(array)):
            array_str += str(array[i])
            if i < len(array) - 1
                array_str += ", "
        print(array_str)

    def execute_sorting_tests(self):
        print("Running some sorters...")
        print("==========================")

        array = [4, 3, 10, 1, 5, 2, 2, 8, 23, 2, 8, 13]

        sorter = BubbleSorter(array)
        sorter.sort()
        sorted_array = sorter.array
        self.print_array(sorted_array)

        sorter = CustomSorter(array)
        sorter.sort()
        sorted_array = sorter.array
        self.print_array(sorted_array)

        sorter = InsertionSorter(array)
        sorter.sort()
        sorted_array = sorter.array
        self.print_array(sorted_array)

    def execute_performance_tests(self):
        print("\nRunning performance tests...")
        print("==========================")

        test = PerformanceTest()
        test_array_size = 5000
        large_array = [random.randint(0, test_array_size) for _ in range(test_array_size)]
        
        expected_sorted_array = sorted(large_array)

        test.evaluate_performance(BubbleSorter(large_array.copy()), large_array.copy(), expected_sorted_array)
        test.evaluate_performance(SelectionSorter(large_array.copy()), large_array.copy(), expected_sorted_array)
        test.evaluate_performance(InsertionSorter(large_array.copy()), large_array.copy(), expected_sorted_array)


if __name__ == "__main__":
    main = SorterMain()
    main.execute()

File: Problems/FixTheCode/FTC_3_sorting-algorithms/java/source/Main.java
// Program that will have some sort algorithms for integer arrays

class Main {

    public static void main(String[] args) {
        executeSortingTests();
        executePerformanceTests();
    }

    public static void PrintArray(int[] array){
        String arrayStr = "";
        for (int i=0; i<array.length; i++){
           arrayStr += array[i];
           if (i<array.length-1)
               arrayStr += ", "
        }
        System.out.println(arrayStr);
    }

    private static void executeSortingTests(){
        System.out.println("Running some sorters...");
        System.out.println("==========================");

        int[] array = {4, 3, 10, 1, 5, 2, 2, 8, 23, 2, 8, 13};

        Sorter sorter = new BubbleSorter();
        int sortedArray[] = sorter.Sort(array);
        PrintArray(sortedArray);

        sorter = new SelectionSorter();
        int sortedArray[] = sorter.Sort(array);
        PrintArray(sortedArray);

        sorter = new InsertionSorter();
        sortedArray = sorter.Sort(array);
        PrintArray(sortedArray);

    }


    private static void executePerformanceTests(){
        System.out.println("\nRunning performance tests...");
        System.out.println("==========================");

        PerformanceTest test = new PerformanceTest();
        int testArraySize = 50000;
        int[] largeArray = new int[testArraySize];
        for (int i=0; i<largeArray.length; i++){
            largeArray[i] = (int)(Math.random() * testArraySize);
        } 
        int[] expectedSortedArray = largeArray.clone();
        java.util.Arrays.sort(expectedSortedArray);

        test.EvaluatePerformance(new BubbleSorter(), largeArray, expectedSortedArray);
        test.EvaluatePerformance(new SelectionSorter(), largeArray, expectedSortedArray);
        test.EvaluatePerformance(new InsertionSorter(), largeArray, expectedSortedArray);
    }
}